gc.c

```diff
@@ -5481,6 +5492,46 @@ gc_sweep_page(rb_objspace_t *objspace, rb_heap_t *heap, struct gc_sweep_context
     gc_report(2, objspace, "page_sweep: end.\n");
 }
 
+static void
+gc_sweep_page(rb_objspace_t *objspace, rb_size_pool_t *size_pool, struct heap_page *sweep_page, int *unlink_limit, int *swept_slots, bool need_pool)
```

Comment:
I don't think `gc_sweep_page` should take care of moving pages to pooled pages or to the tomb heap. For compaction, we shouldn't need to do either of the two, so I think the logic here should stay in `gc_sweep_step`.

Reply:
Thanks Peter - I've had a go at teasing apart some of these conditionals so now `gc_sweep_page` is responsible for adding the page back to the `free_pages` where needed, but `gc_sweep_step` retains the responsibility for adding the page to the tomb, and the pool pages if necessary. What do you think?